crates/sui-core/src/checkpoints2/mod.rs

```diff
@@ -52,14 +59,42 @@ struct CheckpointStoreTables {
     /// Lists all transaction digests included in checkpoints
     /// This can be cleaned up on epoch boundary
     digest_to_checkpoint: DBMap<TransactionDigest, CheckpointSequenceNumber>,
+
+    /// Stores pending signatures
+    /// The key in this table is checkpoint sequence number and an arbitrary integer
+    /// This tables needs to be cleaned up on epoch boundary
+    pending_signatures: DBMap<(CheckpointSequenceNumber, u64), SignedCheckpointData>,
```

Comment:
What's the reason to store it in db instead of in-memory?

Reply:
If keep them in memory then on restart we won't be able to aggregate signatures into certified checkpoint